schema.graphql

```diff
@@ -115,51 +162,98 @@ type DecimalValue @entity {
+  notionalVolTransferredUSD: BigDecimal! # Claim Transferred: Current Underlying and/or Exercise Asset; Option Transferred: Underlying Asset
+  notionalVolSettledUSD: BigDecimal! # Fee Earning Volume; Written + Exercised
+  notionalVolSumUSD: BigDecimal! # Written + Exercised + Redeemed + Transferred
+  notionalVolFeesAccruedUSD: BigDecimal!
+  notionalVolFeesSweptUSD: BigDecimal!
+
+  # Clearinghouse pointer
+  clearinghouse: Clearinghouse!
+  # Metrics for each Token on same day
+  tokenDayDatas: [TokenDayData!]! @derivedFrom(field: "clearinghouseDayData")
 }
 
 type Token @entity {
+  # Token Address
   id: ID!
+
+  # Token Info
   symbol: String!
   name: String!
-  decimals: BigInt!
-  totalValueLocked: BigDecimal!
-  totalValueLockedUSD: BigDecimal!
+  decimals: Int!
+
+  # Total number of tokens in Clearinghouse
+  totalValueLocked: BigInt!
+
+  # Amount of tokens that are ready to be swept as fees
+  feeBalance: BigInt!
+  # Lifetime sum of tokens paid to Valorem as fees
+  feesAccrued: BigInt!
+
+  # Daily metrics (tvl, volume, etc) for the token
+  tokenDayDatas: [TokenDayData!]! @derivedFrom(field: "token")
 }
 
 type TokenDayData @entity {
-  # token address concatendated with date
+  # Token Address + Timestamp of day at 00:00AM UTC
   id: ID!
-  # timestamp rounded to current day by dividing by 86400
+  # Timestamp of day at 00:00AM UTC
   date: Int!
-  # pointer to token
+
+  # The Token that these metrics belong to
   token: Token!
-  # tvl across all contracts in token units
-  totalValueLocked: BigDecimal!
-  # tvl across all contracts in usd
-  totalValueLockedUSD: BigDecimal!
-  # token volume in token units
-  volume: BigDecimal!
-  # token volume in usd
-  volumeUSD: BigDecimal!
+
+  # TVL in Clearinghouse in token units
+  totalValueLocked: BigInt!
+
+  ## Notional Volumes in token units
+  notionalVolWritten: BigInt! # Underlying Asset written
+  notionalVolExercised: BigInt! # Exercise Asset exercised for Options
+  notionalVolRedeemed: BigInt! # Underlying and/or Exercise Asset redeemed from Claims
+  notionalVolTransferred: BigInt! # Claim Transferred: Current Underlying and/or Exercise Asset; Option Transferred: Underlying Asset
+  notionalVolSettled: BigInt! # Fee Earning Volume; Written + Exercised
+  notionalVolSum: BigInt! # Written + Exercised + Redeemed + Transferred
+  notionalVolFeesAccrued: BigInt!
+  notionalVolFeesSwept: BigInt!
+
+  ## Notional Volumes in USD
+  notionalVolWrittenUSD: BigDecimal!
+  notionalVolExercisedUSD: BigDecimal!
+  notionalVolRedeemedUSD: BigDecimal!
+  notionalVolTransferredUSD: BigDecimal!
+  notionalVolSettledUSD: BigDecimal!
+  notionalVolSumUSD: BigDecimal!
+  notionalVolFeesAccruedUSD: BigDecimal!
+  notionalVolFeesSweptUSD: BigDecimal!
+
+  # Pointer to ClearinghouseDayData for same day
+  clearinghouseDayData: ClearinghouseDayData!
 }
 
-type FeeSwitch @entity {
-  # OptionSettlementEngine contract address
+type Clearinghouse @entity {
```

Comment:
Let's just call it OptionSettlementEngine, unfortunately had a flip flop when editing content, because it's a clearinghouse and underwriting system, and token uri generator, and entropy pool, composing an engine that settles options. @neodaoist @Flip-Liquid @nickadamson

Reply:
`type Enigma @entity {}`
jk, you got it :+1: 
changed here https://github.com/valorem-labs-inc/valorem-subgraph/pull/43/commits/b36bf660033804633096bbae936217ca385c424b